src/bproxy.c

```diff
@@ -261,19 +271,20 @@ void read_cb(uv_stream_t *handle, ssize_t nread, const uv_buf_t *buf)
 
       proxy_http_request(ip_port.ip, ip_port.port, conn);
     }
-    else if (conn->proxy_handle)
-    {
-      write_buf(conn->proxy_handle, buf->base, nread);
-    }
   }
   else
   {
     if (nread != UV_EOF)
     {
       log_error("could not read from socket!");
     }
-    if (uv_is_closing((const uv_handle_t *)handle))
+    if (!uv_is_closing((const uv_handle_t *)handle))
     {
+      if (conn->proxy_handle != NULL && !uv_is_closing((uv_handle_t *)conn->proxy_handle))
```

Comment:
I think that shorter version to check if the value is different than `NULL` is just
```c
if (conn->proxy_handle && !uv_is_closing((uv_handle_t *)conn->proxy_handle))
```
instead of specifically defining comparision value NULL.